src/storage/mongodb/SettingStorage.ts

```diff
@@ -304,6 +304,50 @@ export default class SettingStorage {
     }
   }
 
+  public static async saveCryptoSettings(tenantID: string, cryptoSettingToSave: KeySetting): Promise<void> {
+    // Build internal structure
+    const settingsToSave = {
+      id: cryptoSettingToSave.id,
+      identifier: 'crypto',
+      lastChangedOn: new Date(),
+      content: {
+        crypto: cryptoSettingToSave.crypto
+      },
+    } as SettingDB;
+    // Save
+    await this.saveSettings(tenantID, settingsToSave);
+  }
+
+  public static async getCryptoSettings(tenantID: string): Promise<KeySetting> {
+    // Get the Crypto Key settings
+    const settings = await SettingStorage.getSettings(tenantID,
+      { identifier: TenantComponents.CRYPTO },
+      Constants.DB_PARAMS_MAX_LIMIT);
+    if (settings.count > 0) {
+      const cryptoSetting = {
+        formerKey: settings.result[0].content.crypto.formerKey && settings.result[0].content.crypto.formerKey,
+        formerKeyProperties: {
+          blockCypher: settings.result[0].content.crypto.formerKeyProperties && settings.result[0].content.crypto.formerKeyProperties?.blockCypher,
```

Comment:
You test for `settings.result[0].content.crypto.formerKeyProperties` before the assignation, so `settings.result[0].content.crypto.formerKeyProperties?.blockCypher` does not need to test it again (the ? is a typescript syntactic sugar for that test). 

But the correct test before that variable value assignation is: 

`settings.result[0].content.crypto.formerKeyProperties?.blockCypher && settings.result[0].content.crypto.formerKeyProperties.blockCypher`

The same goes for all variable value assignation here: the interface that define the data structure make some fields optional, so testing for their presence is a requirement before using it.